From 36d90c9895cf87aca6147b8a62a37df0b7600f39 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Thu, 20 Mar 2025 15:32:27 +0100
Subject: [PATCH 1651/1695] clk: stm32mp25: don't put all flexgen clocks as
 critical

No longer necessary to mark all the flexgen clocks as critical for the
cortex-M33 because it now uses the SCMI protocol.

Only keep as the critical the root clocks (flexgen 0 to 6), flexgen 59
for GPU sub system and flexgen 63 for DVFS support (CA35 bypass).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ee41a229cd308e4bed2d88d0d122407dbe006433 clk: stm32mp25: don't put all flexgen clocks as critical]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ib6e2d6a1a864c847518ec45edc8649461aa9dad1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/445674
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/448891
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/clk/clk-stm32mp25.c | 32 +++++++++++---------------------
 1 file changed, 11 insertions(+), 21 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index b6524b967..6f675969f 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -4550,29 +4550,12 @@ static struct clk *stm32mp25_clk_provided[STM32MP25_ALL_CLK_NB] = {
 	[CK_MSI_KER]		= &ck_msi_ker,
 };
 
-static void clk_stm32_set_flexgen_as_critical(void)
-{
-	uint32_t i = 0;
-
-	for (i = 0; i < XBAR_CHANNEL_NB; i++) {
-		unsigned long clock_id = CK_ICN_HS_MCU + i;
-		struct clk *clk = NULL;
-
-		if (!stm32_rcc_has_access_by_id(i))
-			continue;
-
-		clk = stm32mp_rcc_clock_id_to_clk(clock_id);
-		assert(clk);
-
-		clk_enable(clk);
-	}
-}
-
 static bool clk_stm32_clock_is_critical(struct clk *clk)
 {
 
 	struct clk *clk_criticals[] = {
 #ifdef CFG_STM32_CM33TDCID
+		&ck_flexgen_63,
 		&ck_icn_p_gicv2m
 #else
 		&ck_hsi,
@@ -4580,6 +4563,16 @@ static bool clk_stm32_clock_is_critical(struct clk *clk)
 		&ck_msi,
 		&ck_lsi,
 		&ck_lse,
+		&ck_icn_hs_mcu,
+		&ck_icn_ls_mcu,
+		&ck_icn_sdmmc,
+		&ck_icn_ddr,
+		&ck_icn_display,
+		&ck_icn_hsl,
+		&ck_icn_nic,
+		&ck_icn_vid,
+		&ck_flexgen_59,
+		&ck_flexgen_63,
 		&ck_cpu1,
 		&ck_icn_p_syscpu1,
 		&ck_icn_s_ddr,
@@ -5027,9 +5020,6 @@ end:
 		}
 	}
 
-	/* TEMPORARY: Enables all root clocks (could be used by M33) */
-	clk_stm32_set_flexgen_as_critical();
-
 	return TEE_SUCCESS;
 }
 
-- 
2.47.2

